### DISCUSSION TITLE
Angular Universal (SSR) integration. Flash-of-Unstyled-Content

### DISCUSSION THREAD

**Discussion by SimonStiph** [tailwindlabs/tailwindcss#10338] [Help] [answered]
I have Tailwind in an SSR (Angular Universal) setup. It works fantastic except for one issue, it's introduced a flash-of-unstyled-content problem because the injected CSS into the HTML includes all the correct CSS except the Tailwind-generated CSS. It's a HUGE drag on the PageSpeed CLS score because the external stylesheet later loads with the referenced Tailwind CSS.

How can I configure Angular and Tailwind to output to the CSS into the SSR HTML?

**Comment by SimonStiph** [comment, marked as answer]
Found that this is an issue related to how some Angular Universal-based projects are configured.  No need to discuss here further.

**Reply by chedched** [reply]
Do you mind sharing your solution?

**Reply by SimonStiph** [reply]
Sure, but I'm going off of memory now since that was a while ago. I think there was an issue with my deploy-url or base-href setting. If you look at how the Angular-Tailwind builder code works, it needs the base-href setting to match the complete URL of the page for the style to be injected, or something like that.

After I updated the values (I _think_ the base-href was the issue), then the styles were injected properly. Otherwise, the styles do not get injected into the HTML, rather the style.css must load for the styles to work properly.

Hopefully that helps a bit. Best of luck

**Reply by chedched** [reply]
Thank you Simon, I highly appreciate it.
